packages/cli/src/lib/snapshot/snapshotFacade.ts

```diff
@@ -0,0 +1,58 @@
+import {green} from 'chalk';
+import {cli} from 'cli-ux';
+import {Configuration} from '../config/config';
+import {
+  SnapshotOperationAbort,
+  SnapshotSynchronizationAmbiguousMatchesError,
+  SnapshotSynchronizationUnknownError,
+} from '../errors/snapshotErrors';
+import {Snapshot} from './snapshot';
+import {SynchronizationPlan} from './synchronization/synchronizationPlan';
+
+export class SnapshotFacade {
+  public constructor(private snapshot: Snapshot, private cfg: Configuration) {}
+
+  public async tryAutomaticSynchronization() {
+    const plan = await this.createSynchronizationPlan();
+    await this.waitForConfirmation();
+    await this.applySynchronizationPlan(plan);
+  }
+
+  private async waitForConfirmation() {
+    const canApplySynchronizationPlan = await cli.confirm(
+      'Synchronization plan matched all resources with great confidence, do you want to proceed? (y/n)'
```

Comment:
Maybe I'd loop in @jpmarceau  for the wording... 

I think I could ask myself, as a end user, what "great confidence" means... Like, is it necessary ? Could we just say "Synchronization plan matched all resources, do you want to proceed?

Reply:
That's a good question. I don't completely remember how the migration api evaluates this, but Olivier's recommendation sounds good to me :)